### DISCUSSION TITLE
Pricing information

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#92] [Q&A]
Although this is not mentioned anywhere in the docs (to my knowledge), I saw [this comment](https://github.com/retypeapp/retype/discussions/62#discussioncomment-971938) that made me realize there will be a Pro version in the (near?) future.

While I appreciate the statement that says "there will always be a free version", I wonder what will the limitations of this free version be, and if they will deteriorate over time to "attract" more paying customers.

I guess the above question is hard to answer at this point, so I want to ask a different one:

Do you already know what the pricing model will be, and what is the ballpark amount you will charge?

1. Will it be a subscription model, Adobe-style (which I find appalling in the self-hosted software industry)?
2. Will it be a one time fee that is valid until the next major release, SublimeText-style?
3. Will it be a subscription model for a SaaS service, GitBook-style (which is an obvious inspiration to retype)?

I think it is important to clarify these intentions as soon as possible - I would hate to be in a situation where I invest effort in converting my docs to retype, and realize later that the pricing options are out of my budget, or not to my liking.

**Comment by geoffreymcgill** [comment]
Yes, there will be a **Retype Pro** and maybe someday a **Retype Enterprise**. 

Currently, Retype has a 100-page limit. We did a lot of research and it was rare to find any doc type site that went beyond 50 pages, so the 100-page limit should cover ~97% of sites out there.

There are no other limitations. 

Our feeling is that if someone has a Retype project with >100 pages, then Retype is bringing some serious value to the project. That value Retype is delivering is either worth something, or it isn't. If it isn't, then we need to try harder to deliver extra value, and that project is free to use any of the other excellent solutions out there or build their own.

The beauty of Retype is that YOU own and control ALL the data and content (as Markdown files). Retype is a configuration-only solution. Your data is not locked up in some proprietary system. You don't need to export or backup your data from a system that may or may not have a self-hosted backup option. You don't need to worry about Retype going offline and bringing your site down with it. The Retype project/website could completely disappear tomorrow and have zero impact on your website. You can move to another generator system, sometimes within minutes, and certainly within a day. In the future, if a better solution is available, you already have your data and can migrate in a snap.

Another important point: The Retype App (CLI) is completely self-contained and does not make any remote calls to some validation or licensing service. There's nothing we can do to remotely disable or somehow _turn off_ Retype. You're autonomous and anonymous to us. We don't even control the download availability, as Retype is installed directly from package services such as NPM and NuGet, which cannot ever be removed by us.

> I wonder what will the limitations of this free version be

100 pages max. That's the only limitation and I don't see any intention of changing that.

**Retype Pro** will only ADD features. The first two planned extra value features for Retype Pro will be (1) a much higher page limit, maybe unlimited, and (2) the option to remove the `Powered by Retype` branding.

We are planning to offer some add-on services that will have a real time or usage cost to us, so those will likely be **Pro** only.

Another **Pro** feature might be advanced build and runtime analytics, which is something a general user is likely uninterested in or would make no difference to them. Retype currently includes some basic build info, and we will be adding much more in the coming releases. The advanced build analytics is much larger project and will take some time to figure it all out.

> and if they will deteriorate over time to "attract" more paying customers.

Nope. Just the opposite.

New features will be constantly added to all versions and there is absolutely no intention of rug pulling features. That's not how we do business. We will just have to demonstrate with our actions over time.

The 100-page limit and `Powered by Retype` branding are good limits IMHO. Theoretically, if both versions had the exact same features except a **Pro** version removed those two limitations, that covers a lot of scenarios and delivers extra value for those projects that require the extra flexibility.

> 1. Will it be a subscription model, Adobe-style (which I find appalling in the self-hosted software industry)?

I really dislike Adobe's subscription model, so we won't be going anywhere near that.

> 2. Will it be a one time fee that is valid until the next major release, SublimeText-style?

I just happen to be drafting this response in Sublime right now. In my paid and registered install of Sublime. 👍 

Yes, the Sublime model or something like it is what we had in mind.

What do you think a fee/model should be?

I see GitBook has a minimum price of $32/month prepaid for one year ($384/year) for up to 5 users. I think Readme is much higher than that. Some other legacy old-skool systems I've seen have pricing models that are comically high for what they're offering, and Retype will eat their lunch (eventually).

> 3. Will it be a subscription model for a SaaS service, GitBook-style (which is an obvious inspiration to retype)?

I don't think so. GitBook is a hosted solution. Retype is a self-hosted solution and the SaaS hosted site model doesn't really work well for what Retype is. If Retype ever offered a hosted solution, then some kind of premium SaaS model for the hosting might have to be the plan. It does scale well.

**Pro Tip:** You can use GitBook to edit your content within their online system, and they offer the option to save edits directly to a GitHub repo you control. Retype fully supports the `.md` files created by GitBook, including the `SUMMARY.md` navigation configuration. Then you can use Retype to self-host the project instead of having GitBook host. Retype includes many more features and components than GitBook, so you would be limited (by GitBook) with this scenario, but it is possible and fun to see working.

--- 

So, we're serious about what we're building, we focus on the details, performance is a priority one, and we take great pride in the quality of our work. We feel we're building something totally awesome and delivering real value to users. That comes with real costs, real families to support, and it would be nice to grow the team. That all takes some serious dollarydoos. 

Hopefully **Retype Pro** can help recover some of the enormous investment we have already made into Retype and help us grow. There's so much more we wish we could be doing right now, but resource limitations (time & money) are holding us back. That said, we're building as fast as we can and improving Retype daily.

Hope this helps answer your questions. Any feedback is greatly appreciated. Peace ✌️.

**Reply by geoffreymcgill** [reply]
I initially planned to write a quick response. That turned into a wall of text. Probably didn't need to write that much.

**Reply by DannyBen** [reply]
> That turned into a wall of text

Nevertheless, I plan on reading it carefully.

**Reply by DannyBen** [reply]
So, first of all, thanks for the detailed and honest answer.

I am already familiar with the 100 page limitation (which I agree, sounds generous).

> The beauty of Retype is that YOU own and control ALL the data and content

I am completely aware of this, and it is a mandatory requirement for me. Just as a background, until two days ago - when I discovered retype - my go to tool was mdbook for that same reason.

Prior to mdbook, I used GitBook, because its design was more to my liking than anything else, but if I am being honest, I was not crazy about their approach to certain things, and did not like the fact that it is not running on my own host of choice, and local environment.

In that regard, Retype is like the holy grail for documentation generators.

> Another important point: The Retype App (CLI) is completely self-contained

This is another point I love about Retype. Usually I try to avoid `npm install <anything>` as much as possible, since most packages bring thousands upon thousands of `node_modules` for no apparent reason, but installing and running the Retype CLI is completely frictionless.

> The advanced build analytics is much larger project and will take some time to figure it all out

I wonder - why should anyone care about build analytics, *especially* if the level of effort to develop it is high? Or perhaps I don't understand what it means? If it means metrics about the site build process, I can't think of a reason why anyone should care - enterprise user or not.

>  there is absolutely no intention of rug pulling features

When I posted the question, I was less familiar with how you conduct yourself. After the past 24 hours of discussions and issues, I know better.

> The 100-page limit and Powered by Retype branding are good limits IMHO

I totally agree. As a (currently non-paying) user, I can definitely live with these two, and I even wear the `Powered by Retype` badge with pride, and would probably not remove it even if I happen to need the Pro version. 

Now, to look at the same two limitations from another angle - as a user who wants to see Retype succeed, evolve, and stay in business - I am a little concerned that these two limitations might not attract sufficient paying users. I very much hope I am wrong, but if I am not then Retype will face a dilemma:

- Either be more aggressive in the pro vs free differentiation - and potentially start to alienate some of the free users (see what DockerHub is doing with "no more auto builds for free users").
- Or stick to your "product" guns, and guard the interests of the free users, which might lead to everybody in Retype starting to lose interest, because it does not bring in the desired amount of income.

> I really dislike Adobe's subscription model, so we won't be going anywhere near that.
> ...
> Yes, the Sublime model or something like it is what we had in mind.

Me too, and I am happy to hear. Out of all pricing models I mentioned, for me, the only one I can accept in regards to products like Retype, is a one time payment that grants me usage rights until the next major version, or something like this. I am also a happy paying Sublime Text customer, and was more than happy to pay for the upgrade from 2 to 3 - it was long awaited, it was reasonably priced, and it packed **tons** of value to me.

> I see GitBook has a minimum price of $32/month prepaid for one year 

This is outrageous. As someone who works a lot with cloud services - both as a hobby and in my day to day work - I know how much hosting costs. This is completely out of proportion, and actually, it is one of the reasons that prompted me to ask about the Retype pricing.

Since Retype seems to be **very** influenced by the GitBook design (are there ex-gitbook employees on the Retype team?...) - I was concerned that it is a "spinoff" of GitBook, and the same pricing practices will soon creep in.

> Pro Tip: You can use GitBook to edit your content within their online system

Thanks, but "online editing" is actually for most of my purposes, a downside.
I enjoy the command line, and I enjoy Sublime Text. 

While GitBook is providing both online editing and markdown upload, one might say that it is **not** a downside, but I claim that it is. It means that the GitBook people are focusing on things that I do not care about. This is one of the things that sit like a glove for me with Retype - it does **exactly** what I need. From the simple command line, to the pitch-perfect design, without assuming anything about me or about what I need, but still with an opinionated (and a good opinion at that) approach to everything.

> So, we're serious about what we're building, we focus on the details, performance is a priority one, and we take great pride in the quality of our work.

It is unarguably evident in both the product and in these fun discussions we have had in the past 24 hours. You might think I am saying this to everyone, but people who know me, know that I am criticizing everything and am generally unhappy with quality of things. Everything I know about Retype so far is just perfect.

> Hopefully Retype Pro can help recover some of the enormous investment we have already made

Assuming a "software style" pricing model, even though I do not need any of the Pro features, I am happy to be your first paying customer when it happens as a show of support. Quality software deserves monetary compensation.

Thanks for listening, and sorry for my own wall of text....

**Reply by geoffreymcgill** [reply]
> Since Retype seems to be very influenced by the GitBook design (are there ex-gitbook employees on the Retype team?...) - I was concerned that it is a "spinoff" of GitBook, and the same pricing practices will soon creep in.

Ha, no. There is no history between Retype and GitBook.

I'm somewhat uncertain how much exactly of Retype was influenced by GitBook. A few minor bits of the UI maybe, but I can't think of anything specifically. Most of what we have built in the UI are standard elements that just make for a good product.

Believe it or not, if I was developing the Retype UI in a vacuum, and had never seen GitBook, Retype would still have likely resulted in the same generated UI as what you see today.

Retype (and GitBook) just have good clean UI's. When you start breaking down the requirements for displaying the content and keep removing things that don't need to be there, you end up with a clean UI like Retype or GitBook.

The seed of Retype was planted more than 15 years ago. We made a few attempts over the years to kick-off the project, but we always had something else forcing our attention. I discovered GitBook a few years ago and my initial thought was _"this looks great, now I don't have to build it"_. I was happy. We tried GitBook a few times, but it just never exactly met our requirements and there were always compromises. We repeatedly hit a wall and eventually decided to build something that met our requirements. 

We just needed more features, self-hosting, local runtime, better performance, etc.

GitBook has done an awesome job with their product and it looks like they have a super talented team. I think they're based in France too? I like France.

Hope this helps.

**Reply by geoffreymcgill** [reply]
I should clarify. We certainly wanted to reproduce much of the _features_ of GitBook, as well as several other similar products. Although again, even if none of those other products existed, we likely would have still ended up in the same place with Retype, feature wise.

**Reply by DannyBen** [reply]
Yes. It seems like Retype is the culmination of feature maturity + unsatisfied product needs.

**Reply by olivatooo** [reply]
How to bypass this 100 page limit? :thinking:

**Reply by gtnardy** [reply]
> 
> 
> How to bypass this 100 page limit? 🤔

I would love to know! As I can't find pricing for Pro version as well, my project is hitting 100 files now lol.
But honestly 100 files doesn't seem a big project :/

**Reply by geoffreymcgill** [reply]
Please send an email to me at hello@retype.com and I will send back a temp key to unlock the 100 page limit.

We are still working on the **Retype Pro** model, which is scheduled for a late September release. Pricing will be very reasonable. I will be sharing our thoughts regarding the model/pricing and soliciting feedback in early September.\

**UPDATE**: Retype Pro is now [available](https://retype.com/pro/).

**Reply by olivatooo** [reply]
For anyone who is looking, try docsaurus, it's free and don't have these limitations

**Reply by geoffreymcgill** [reply]
@olivatooo Docusaurus is an excellent solution. We would highly recommend everyone check it out.

Another option to investigate would be [MkDocs](https://www.mkdocs.org/).

**Reply by eslachance** [reply]
We're now mid October and there's still no indication of pricing or licensing other than the vague reference to the Pro version in the roadmap, and the hidden 100 page limit that seems undocumented everywhere. 

Maybe *at least* document this limitation somewhere so that people aren't surprised when it strikes out of nowhere?
![image](https://user-images.githubusercontent.com/643344/137320931-474bdd5d-b3cb-403f-a671-f9c57bba527e.png)

**Reply by DannyBen** [reply]
Why poke the bear?... :)

**Reply by geoffreymcgill** [reply]
> Maybe _at least_ document this limitation somewhere so that people aren't surprised when it strikes out of nowhere?

Agreed. We should have documented the 100 page limit. I will get on that.

Retype Pro licensing will be available as of `v1.11`, which we're working on right now. It's a big job figuring all this out and we're a small team, so unfortunately the process is taking a little longer than we had hoped.

If you need a >100 page limit before `v1.11` is released, please send me an email (hello@retype.com) with the request and I can respond with an option to remove the limit.

We are drafting the Retype Pro licensing model proposal which will be posted here for community review and feedback before we commit to anything.

Hope this helps.

**UPDATE**: Retype Pro is now [available](https://retype.com/pro/).

**Reply by dwbelliston** [reply]
Project is amazing... just migrated away from mkdocs in a day. Deleted lots of config code to make mkdocs look nice. This is pretty out of the box and way simpler to setup up. Love to see free version will be around so i can use this for my open source project!

**Reply by geoffreymcgill** [reply]
Thanks for the feedback @dwbelliston! 👍 

> Love to see free version will be around so i can use this for my open source project!

Nothing is changing with the free version of Retype.

Retype Pro will unlock a greater page limit. Currently the limit is 100 pages max.

BTW, if you need to build more >100 pages, just send me an email (hello@retype.com) and I can send back a temp key to unlock. 

Hope this helps.

**UPDATE**: Retype Pro is now [available](https://retype.com/pro/).

**Comment by LoneDev6** [comment]
The 100 pages limit really needs to be documented in the top part of the Retype welcome page instead of being hard to find

**Reply by RickCogley** [reply]
Saw Pro mentioned in the release notes and would love to know more :-)

**Reply by geoffreymcgill** [reply]
Retype Pro is now [available](https://retype.com/pro/).

**Reply by markbattistella** [reply]
@geoffreymcgill sorry to rehash an old/ongoing discussion but I'm a little confused with the "Websites – max 100 pages each" vs "Websites – max 1000 pages each" part. 

In this bit:
![image](https://user-images.githubusercontent.com/17962160/162563308-0c17eb69-2a1e-44ed-b480-adbafaa89013.png)

What does "unlimited" mean. Is that unlimited 100 pages per directory, or in total or generated? I'm getting thrown because unlimited means that there's infinite amount, so infinite amount of 100 pages?

Does that also mean that the README is counted, and if there's a TOS, PP, Cookies page too? 

And how does it affect Blogs? Is it 100 sidebar pages, but blog 100 new pages, or is it a collective?

**Reply by geoffreymcgill** [reply]
Hi @markbattistella. The "unlimited" would be any number of projects, each with a max of 100 pages.

Each `.md` file is considered a page. Nothing else counts, just `.md` files. 

Blog pages are treated just as any other `.md` file/page, so they count towards to the max page limit.

I'll put some time into reviewing that table with the "Unlimited" labels and see if there's a better way to document and describe the limits.

**Reply by markbattistella** [reply]
Thanks @geoffreymcgill that makes sense now..

Maybe this might be more in line of what you were intending:
![Screen Shot 2022-04-10 at 11 53 25 am](https://user-images.githubusercontent.com/17962160/162597856-67c6cd86-1510-4ccb-b01e-fcdaea01bce2.png)

**Comment by geoffreymcgill** [comment]
A **Retype Pro** proposal is made in #224.
